### PR TITLE
created diagram_cim_writer and diagram_server_writer

### PR DESCRIPTION
**Pull request by Esaotuelr**
Signed-off-by: Jimmy Tung <jimmy.tung@zepben.com>

### CODE REVIEW

**Comment by ryanjcoleman** [comment]
Task linked: [EWB-586 Support writing database](https://app.clickup.com/t/6929263/EWB-586)

File: src/zepben/evolve/database/sqlite/writers/diagram_cim_writer.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        insert.add_value(table.ORIENTATION_KIND.queryIndex, diagram.orientationKind.name)
+        return self.save_identified_object(table, insert, diagram, "diagram")
+
+    def save_diagram_object(self, diagram_object: DiagramObject) -> bool:
+        table = self.databaseTables.getTable(TableDiagramObjects)
+        insert = self.databaseTables.getInsert(TableDiagramObjects)
+        status = True
+        for sequence, point in enumerate(diagram_object.points):
+            status = status and self.save_diagram_object_point(diagram_object, point, sequence) 
+        insert.add_value(table.IDENTIFIED_OBJECT_MRID.queryIndex, diagram_object.identifiedObjectMRID)
```
Comment (kgreav) [inline src/zepben/evolve/database/sqlite/writers/diagram_cim_writer.py:31]: these things are lower case (IDENTIFIED_OBJECT_MRID and such)

File: src/zepben/evolve/database/sqlite/writers/diagram_cim_writer.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        insert.add_value(table.DIAGRAM_MRID.queryIndex, diagram_object.diagram.mRID)
+        insert.add_value(table.STYLE.queryIndex, diagram_object.style)
+        insert.add_value(table.ROTATION.queryIndex, diagram_object.rotation)
+        return status and self.save_identified_object(table, insert, diagram_object, "diagram object")
+
+    def save_diagram_object_point(self, diagram_object: DiagramObject, diagram_object_point: DiagramObjectPoint, sequence_number: int) -> bool:
+        table = self.databaseTables.getTable(TableDiagramObjectPoints)
```
Comment (kgreav) [inline src/zepben/evolve/database/sqlite/writers/diagram_cim_writer.py:38]: this should be `get_table` and `get_insert`

File: src/zepben/evolve/database/sqlite/writers/diagram_cim_writer.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+from zepben.evolve.database.sqlite.writers.base_cim_writer import BaseCIMWriter
+
+
+@dataclass
+class DiagramCIMWriter(BaseCIMWriter):
+
+    # ** ** ** ** ** ** IEC61970 DIAGRAM LAYOUT ** ** ** ** ** ** #
+
+    def save_diagram(self, diagram: Diagram) -> bool:
+        table = self.databaseTables.getTable(TableDiagrams)
```
Comment (kgreav) [inline src/zepben/evolve/database/sqlite/writers/diagram_cim_writer.py:19]: same here - get_table and get_insert, and in the other functions.

File: src/zepben/evolve/database/sqlite/writers/diagram_service_writer.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from typing import Type, TypeVar
+
+from zepben.evolve import NetworkService, IdentifiedObject, DiagramObject, Diagram
+from zepben.evolve.database.sqlite.writers.base_service_writer import BaseServiceWriter
+from zepben.evolve.database.sqlite.writers.network_cim_writer import NetworkCIMWriter
+
+T = TypeVar("T", bound=IdentifiedObject)
```
Comment (kgreav) [inline src/zepben/evolve/database/sqlite/writers/diagram_service_writer.py:12]: can remove this line, that was a mistake from my code

**Review by kgreav — CHANGES_REQUESTED** [review]

**Review by kgreav — APPROVED** [review]

**Pull request merged**